static/js/popover_menus.js

```diff
@@ -185,4 +202,140 @@ export function initialize() {
             compose_enter_sends_popover_displayed = false;
         },
     });
+
+    delegate("body", {
+        ...default_popover_props,
+        // The is our minimum supported width for mobile. We shouldn't
+        // make the popover wider than this.
+        maxWidth: "320px",
+        target: ".actions_hover .zulip-icon-ellipsis-v-solid",
+        placement: "bottom",
+        popperOptions: {
+            modifiers: [
+                {
+                    // The placement is set to bottom, but if that placement does not fit,
+                    // the opposite top placement will be used.
+                    name: "flip",
+                    options: {
+                        fallbackPlacements: ["top"],
+                    },
+                },
+            ],
+        },
+        onShow(instance) {
+            on_show_prep(instance);
+            const $row = $(instance.reference).closest(".message_row");
+            const message_id = rows.id($row);
+            const args = popover_menus_data.get_actions_popover_content_args(message_id);
+            instance.setContent(parse_html(render_actions_popover_content(args)));
+            $row.addClass("has_popover has_actions_popover");
+            message_actions_popover_displayed = true;
+        },
+        onMount(instance) {
+            // We want click events to propagate to `instance` so that
+            // instance.hide gets called.
+            // Copy link to message is independently handled in popovers.js.
+            const $popper = $(instance.popper);
+            $popper.one("click", ".respond_button", (e) => {
+                // Arguably, we should fetch the message ID to respond to from
+                // e.target, but that should always be the current selected
+                // message in the current message list (and
+                // compose_actions.respond_to_message doesn't take a message
+                // argument).
+                compose_actions.quote_and_reply({trigger: "popover respond"});
+                e.preventDefault();
+            });
+
+            $popper.one("click", ".popover_edit_message, .popover_view_source", (e) => {
```

Comment:
Is there a good way to check that these get garbage-collected when the instance is destroyed?

Reply:
We remove the popover from DOM when it is hidden via `instance.destroy` and all the events here are attached to the popover rather than `body`. So, since these event handlers have no reference once the popover is removed from DOM, browser automatically remove these event handlers.

I couldn't find a way to verify that this is happening locally, though.

Reply:
Makes sense; the browser "event handlers" tab I think requires an element to see the event handlers attached to the element, and so there's no way to check directly. Sounds good.